client/modules/User/actions.js

```diff
@@ -275,16 +275,26 @@ export function submitSettings(formValues) {
 
 export function updateSettings(formValues) {
   return (dispatch) =>
-    new Promise((resolve) =>
+    new Promise((resolve) => {
+      if (!formValues.currentPassword && formValues.newPassword) {
+        dispatch(showToast(5500));
+        dispatch(setToastText('Toast.EmptyCurrentPass'));
+        resolve();
+        return;
+      }
+
       submitSettings(formValues)
         .then((response) => {
           dispatch(updateSettingsSuccess(response.data));
           dispatch(showToast(5500));
           dispatch(setToastText('Toast.SettingsSaved'));
           resolve();
         })
-        .catch((error) => resolve({ error }))
-    );
+        .catch((error) => {
```

Comment:
I wonder if we can look here and check if it's specifically a 401 error so that we know it's an incorrect password.  If it's a 500 we could show a generic "something went wrong".  We could also think about using the message from the server which plays into a larger discussion on translating server-side messages.

Reply:
Yes, that sounds like a good approach. Differentiating between specific HTTP status codes like : 

- 401 for incorrect passwords 
- 500 for generic errors 

On it